tutorial/markdown/java/spring-boot.md

```diff
@@ -53,172 +95,364 @@ mvn package
+- It receives the airline data in the request body, which should be a valid JSON representation of an airline.
+- It calls the `createAirline` method of the `AirlineService` to create the new airline.
+- If the airline is created successfully, it returns a `ResponseEntity` with HTTP status `201 Created` and the created airline data in the response body.
+- If a conflict occurs (e.g., an airline with the same ID already exists), it returns a `ResponseEntity` with HTTP status `409 Conflict`.
+- If any other error occurs, it returns a `ResponseEntity` with HTTP status `500 Internal Server Error`.
 
-Update a Profile by Profile ID
+### PUT Route
 
-Now let's navigate to the update method of the ProfileController class.  The entire document gets replaced except for the document key and the `pid` field.  We create a call to the `collection` using the `upsert` method and then return the document saved and the result just as we did in the previous endpoint.
+`@PutMapping("/{id}")`
+This route is used to update an existing airline by its ID.
 
-```java
-profileCol.upsert(id, profile);
-return ResponseEntity.status(HttpStatus.CREATED).body(profile);
-```
+- It expects the id of the airline as a path parameter.
+- It receives the updated airline data in the request body.
+- It calls the `updateAirline` method of the `AirlineService` to update the airline with the specified ID.
+- If the airline is updated successfully, it returns a `ResponseEntity` with HTTP status `200 OK` and the updated airline data in the response body.
+- If the airline with the specified ID is not found, it returns a `ResponseEntity` with HTTP status `404 Not Found`.
+- If any other error occurs, it returns a `ResponseEntity` with HTTP status `500 Internal Server Error`.
+
+### DELETE Route
 
-> *from update method of controllers/ProfileController.java*
+`@DeleteMapping("/{id}")`
+This route is used to delete an airline by its ID.
 
-## DELETE Profile
+- It expects the id of the airline as a path parameter.
+- It calls the `deleteAirline` method of the `AirlineService` to delete the airline with the specified ID.
+- If the airline is deleted successfully, it returns a `ResponseEntity` with HTTP status `204 No Content` (indicating success with no response body).
+- If the airline with the specified ID is not found, it returns a `ResponseEntity` with HTTP status `404 Not Found`.
+- If any other error occurs, it returns a `ResponseEntity` with HTTP status `500 Internal Server Error`.
 
-Navigate to the delete method in the ProfileController class.  We only need the `Key` or id from the user to remove a document using a basic key-value operation.
+These routes together provide the basic CRUD (Create, Read, Update, Delete) operations for managing airlines via the RESTful API. The `AirlineService` contains the business logic for these operations, and the `AirlineRepository` interacts with the database to perform the actual data operations.
+
+## Route Workflow
+
+### GET Route Workflow
```

Comment:
Can you combine this with the description in the above section?

Reply:
Now they are one section